product_supplierinfo_import_by_barcode/models/product_supplierinfo_import_template.py

```diff
@@ -8,13 +8,31 @@ class ProductSupplierInfoImportTemplate(models.Model):
     _description = "Templates to be detected when importing vendor pricelist"
 
     name = fields.Char(required=True)
+    supplier_id = fields.Many2one(comodel_name="res.partner")
     header_offset = fields.Integer(required=True, default=0)
-    barcode_header_name = fields.Char(required=True)
+    sheet_index = fields.Integer(required=True, default=0)
```

Comment:
UX suggestion: for a regular user isn't very intuitive that the first sheet is the number 0. 
```suggestion
    sheet_number = fields.Integer(required=True, default=1, help="The number of the sheet page in the spreadsheet where the pricelist are")
```